android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/WifiDirectConnect.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.physical_web.physicalweb;
+
+import org.physical_web.collection.UrlDevice;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.NetworkInfo;
+import android.net.Uri;
+import android.net.wifi.p2p.WifiP2pConfig;
+import android.net.wifi.p2p.WifiP2pInfo;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.net.wifi.p2p.WifiP2pManager.ActionListener;
+import android.net.wifi.p2p.WifiP2pManager.Channel;
+import android.util.Log;
+
+/**
+ * This class is for using WifiDirect to create a WifiDirect
+ * P2P connection with the Physical Web Device.
+ */
+class WifiDirectConnect {
+  private static final String TAG = "WifiDirectConnect";
+  private static Context mContext;
+  private static WifiP2pManager mManager;
+  private static Channel mChannel;
+  private static WiFiDirectBroadcastReceiver mReceiver = null;
+  private static UrlDevice mDevice;
```

Comment:
Do these need to be static?  I think it would be better to avoid global state for something like this, even if it's not possible to connect to two WifiDirect beacons at once due to hardware limitations.